File: nan21.dnet.module.ad/nan21.dnet.module.ad.presenter/src/main/java/net/nan21/dnet/module/ad/workflow/ds/model/WfDefNodeDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.ad.workflow.ds.model;

import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeDs;

import net.nan21.dnet.module.ad.workflow.domain.entity.WfDefNode;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = WfDefNode.class, sort = { @SortField(field = WfDefNodeDs.fID) })
public class WfDefNodeDs extends AbstractTypeDs<WfDefNode> {

    public static final String fASSIGNTOUSER = "assignToUser";
    public static final String fASSIGNTOGROUP = "assignToGroup";
    public static final String fSTARTWITHPREVIOUS = "startWithPrevious";
    public static final String fTASKTYPE = "taskType";
    public static final String fPROCESSID = "processId";
    public static final String fPROCESS = "process";

    @DsField()
    private String assignToUser;

    @DsField()
    private String assignToGroup;

    @DsField()
    private Boolean startWithPrevious;

    @DsField()
    private String taskType;

    @DsField(join = "left", path = "process.id")
    private Long processId;

    @DsField(join = "left", path = "process.name")
    private String process;

    public WfDefNodeDs() {
        super();
    }

    public WfDefNodeDs(WfDefNode e) {
        super(e);
    }

    public String getAssignToUser() {
        return this.assignToUser;
    }

    public void setAssignToUser(String assignToUser) {
        this.assignToUser = assignToUser;
    }

    public String getAssignToGroup() {
        return this.assignToGroup;
    }

    public void setAssignToGroup(String assignToGroup) {
        this.assignToGroup = assignToGroup;
    }

    public Boolean getStartWithPrevious() {
        return this.startWithPrevious;
    }

    public void setStartWithPrevious(Boolean startWithPrevious) {
        this.startWithPrevious = startWithPrevious;
    }

    public String getTaskType() {
        return this.taskType;
    }

    public void setTaskType(String taskType) {
        this.taskType = taskType;
    }

    public Long getProcessId() {
        return this.processId;
    }

    public void setProcessId(Long processId) {
        this.processId = processId;
    }

    public String getProcess() {
        return this.process;
    }

    public void setProcess(String process) {
        this.process = process;
    }

}

File: nan21.dnet.module.ad/nan21.dnet.module.ad.ui.extjs/src/main/webapp/src/ds/WfDefNodeDs.js
 
   	 
Ext.define("net.nan21.dnet.module.ad.workflow.ds.model.WfDefNodeDs" , {
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"name",type:"string" },
		{name:"active",type:"boolean" },
		{name:"description",type:"string" },
		{name:"id",type:"int", useNull:true },
		{name:"clientId",type:"int", useNull:true },
		{name:"createdAt",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"modifiedAt",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"createdBy",type:"string" },
		{name:"modifiedBy",type:"string" },
		{name:"version",type:"int", useNull:true },
		{name:"assignToUser",type:"string" },
		{name:"assignToGroup",type:"string" },
		{name:"startWithPrevious",type:"boolean" },
		{name:"taskType",type:"string" },
		{name:"processId",type:"int", useNull:true },
		{name:"process",type:"string" }
	],
	validations: [  
		{field: "name", type: 'presence'}, 
		{field: "taskType", type: 'presence'}
	]
});
	
Ext.define("net.nan21.dnet.module.ad.workflow.ds.param.WfDefNodeDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 